### PR TITLE
Feature/catenax wrapper path

### PR DESCRIPTION
**Pull request by drcgjung**
What this PR changes/adds
Currently, the inbound Filter forgets the actual path that is hit under the base-path/deployment-path.
This also leads to the "name" compartment of the HttpDataSource being left empty.
This fix computes the difference between basepath and requestpath and uses it as the "name" property of the DataAddress in the request --> HttpDataSource gets a (hierarchical) name that is appended to the target url.

Why it does that
For wrapping arbitrary REST APIs using the http dataplane, it is necessary not only to take headers, params and body with the data request, but also the concrete URL that is hit (otherwise you would need indefinitely many data planes/addresses to be registered).

Further notes
Tests have been checked before and after (quite some were already red in the main branch, so this was rather a guess than a profound investigation), but no additional test was added (maybe you should think about it).

## Linked Issue(s)

Closes # <-- _insert Issue number if one exists_

## Checklist

- [ ] added appropriate tests?
- [ ] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [ ] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [ ] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [styleguide](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/styleguide.md) for details_)

### CODE REVIEW

**Comment by florianrusch-zf** [comment]
@bscholtes1A I think you wanted to have a look on it or?

**Comment by florianrusch-zf** [comment]
Is related to #881

**Review by ndr-brt — COMMENTED** [review]
This code breaks `DataPlanePublicApiRequestFilterTest` tests.
Please add at least one test that verifies the condition

File: extensions/data-plane/data-plane-api/src/main/java/org/eclipse/dataspaceconnector/dataplane/api/transfer/DataPlanePublicApiRequestFilter.java [hunk 1]
```diff
@@ -119,6 +119,17 @@ public void filter(ContainerRequestContext requestContext) throws IOException {
...
         var dataAddress = typeManager.readValue(claims.getClaims().get(DATA_ADDRESS), DataAddress.class);
+
+        // extract the difference between base and request uri as dataaddress name -> httpdatasource
+        String baseUri = requestContext.getUriInfo().getBaseUri().toString();
+        String requestUri = requestContext.getUriInfo().getRequestUri().toString();
+        // TODO what about arbitrary deployment paths under the base path -> now become part of the name
+        String diffUri = requestUri.substring(baseUri.length());
+        if (diffUri.indexOf("?") >= 0) {
+            diffUri = diffUri.substring(0, diffUri.indexOf("?"));
+        }
```
Comment (MoritzKeppler) [inline extensions/data-plane/data-plane-api/src/main/java/org/eclipse/dataspaceconnector/dataplane/api/transfer/DataPlanePublicApiRequestFilter.java:130]: isn't that exactly what https://javadoc.io/static/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/jakarta/ws/rs/core/UriInfo.html#getPath() delivers - the path part of the URI relative to the base path?

**Comment by bscholtes1A** [comment]
This PR should be closed in favor of #929

**Comment by drcgjung** [comment]
Closed in favour of #929

**Pull request closed without merging**